### PR TITLE
Migrate MenuButton component to docs-ui package

### PR DESCRIPTION
**Pull request by felixhabib**
I had to duplicate the touchable folder into the 'docs-ui' package because I couldn't access it within Braid.
Please let me know if there's a better approach.

### CODE REVIEW

File: packages/docs-ui/src/private/touchable/virtualTouchableRules.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { hitArea } from './hitArea';
+
+export const virtualTouchableRules = {
```
Comment (mrm007) [inline packages/docs-ui/src/private/touchable/virtualTouchableRules.ts:3]:
```suggestion
import type { CSSProperties } from '@vanilla-extract/css';

import { hitArea } from './hitArea';

export const virtualTouchableRules: CSSProperties = {
```

Without it, having `transformers: 'rollout'` is not an error. With the change:

error with transformers

File: packages/docs-ui/README.md [hunk 2]
```diff
@@ -34,5 +35,24 @@ import { LinkableHeading } from '@braid-design-system/docs-ui';
...
+import { MenuButton } from '@braid-design-system/docs-ui';
+
+<MenuButton open={menuOpen} onClick={handleOnClick} />;
+```
+
+#### Props
+
+| props   | value        | description                                                                             |
+| ------- | ------------ | --------------------------------------------------------------------------------------- |
+| open    | `boolean`    | The Menu can either be open or closed. If open, the button will change to a close icon. |
```
[the remark this answers is not in the document]
Comment (felixhabib) [reply packages/docs-ui/README.md]: Oh yep sorry, didn't push

File: packages/docs-ui/README.md [hunk 3]
```diff
@@ -16,7 +16,8 @@ npm install @braid-design-system/docs-ui
 
 ### `LinkableHeading`
 
-A heading wrapped in a link to a hash that is generated based on the heading content, enabling deep links within documentation pages.
+A heading wrapped in a link to a hash that is generated based on the heading content, enabling deep links within
+documentation pages.
```
Comment (mrm007) [inline packages/docs-ui/README.md:20]: Why the line break here?

Comment (felixhabib) [reply packages/docs-ui/README.md:20]: Ahh my bad, thought I got it stopped when I disabled prettier, idk what I did.

**Pull request merged**